serial.c

```diff
@@ -452,6 +490,235 @@ static void Interrupt com_general_isr(void)
 /* ======================================================================== */
 
 
+#ifdef PROTECTED_MODE
+
+/* we do not know the exact size of com_general_isr() function
+   but for sure it's not longer then 2048 bytes */
+#define ISR_SIZE        2048
+
+#if defined( __DJGPP__ )
+
+static void serial_dpmi_get_pvect(int vector, _go32_dpmi_seginfo *info)
+{
+    _go32_dpmi_get_protected_mode_interrupt_vector(vector, info);
+}
+
+static void serial_dpmi_set_pvect(int vector, _go32_dpmi_seginfo *info)
+{
+    _go32_dpmi_set_protected_mode_interrupt_vector(vector, info);
+}
+
+static int serial_dpmi_lock_memory(void)
+{
+    unsigned long dataaddr, codeaddr;
+    __dpmi_meminfo dataregion, coderegion;
+
+    if(__dpmi_get_segment_base_address(_my_cs(), &codeaddr) == 0 &&
+       __dpmi_get_segment_base_address(_my_ds(), &dataaddr) == 0)
+    {
+        coderegion.handle = 0;
+        coderegion.size = ISR_SIZE;
+        coderegion.address = codeaddr + (unsigned long)com_general_isr;
+        dataregion.handle = 0;
+        dataregion.size = sizeof(g_comports);
+        dataregion.address = codeaddr + (unsigned long)g_comports;

```

Comment:
Can these be cast to `intptr_t` or 'uintptr_t` instead of `unsigned long`? Pointer arithmetic via integer types is undefined.